pages/[...slug].vue

```diff
@@ -0,0 +1,138 @@
+<script setup>
+// GQL
+import _get from 'lodash/get'
+import GENERAL_CONTENT_DETAIL from '../gql/queries/GeneralContentDetail.gql'
+
+// Helpers
+// import stripMeapFromURI from "../utils/stripMeapFromURI"
+
+// Helpers
+const { $graphql, $getHeaders } = useNuxtApp()
+
+// ROUTE
+const route = useRoute()
+// to do added these from another file, do I need them?
+const path = route.path.replace(/^\/|\/$/g, '') // trim initial and/or final slashes
+const variables = { path }
+
+// ASYNCDATA
+// TODO if errors check params documentation
+const { data, error } = await useAsyncData(`general-content-${path}`, async () => {
+  const data = await $graphql.default.request(GENERAL_CONTENT_DETAIL, variables)
+  console.log('data', data)
+  return data
+})
+// const { data, error } = await useAsyncData(`general-content-${path}`, async ({ $graphql, params }) => {
```

Comment:
Is this for reference or should be deleted?

Reply:
whoops that was for reference, removed it